src/python/graplctl/graplctl/cli.py

```diff
@@ -0,0 +1,607 @@
+import dataclasses
+import uuid
+
+from typing import List
+
+from mypy_boto3_ec2 import EC2ServiceResource
+import mypy_boto3_ec2.service_resource as ec2_resources
+from mypy_boto3_cloudwatch.client import CloudWatchClient
+from mypy_boto3_cloudwatch.type_defs import MetricTypeDef
+from mypy_boto3_sns.client import SNSClient
+from mypy_boto3_ssm import SSMClient
+from mypy_boto3_route53 import Route53Client
+
+import boto3
+import click
+
+from . import __version__
+from . import dgraph_ops
+from . import docker_swarm_ops
+
+EC2: EC2ServiceResource = boto3.resource("ec2")
+SSM: SSMClient = boto3.client("ssm")
+CLOUDWATCH: CloudWatchClient = boto3.client("cloudwatch")
+SNS: SNSClient = boto3.client("sns")
+ROUTE53: Route53Client = boto3.client("route53")
+
+#
+# main entrypoint for grapctl
+#
+
+
+@dataclasses.dataclass
+class GraplctlState:
+    grapl_region: str
+    grapl_prefix: str
+    grapl_version: str
+
+
+@click.group()
+@click.version_option(version=__version__)
+@click.option(
+    "-r",
+    "--grapl-region",
+    type=click.Choice(docker_swarm_ops.REGION_TO_AMI_ID.keys()),
+    envvar="GRAPL_REGION",
+    help="grapl region to target [$GRAPL_REGION]",
+)
+@click.option(
+    "-p",
+    "--grapl-prefix",
+    type=click.STRING,
+    envvar="GRAPL_CDK_DEPLOYMENT_NAME",
+    help="grapl deployment name [$GRAPL_CDK_DEPLOYMENT_NAME]",
+)
```

Comment:
nitpick: 
in the spirit of https://github.com/grapl-security/issue-tracker/issues/125 could we maybe call this `--grapl-deployment-name` / `-n` or something?

(both verbiages are used here)